backend/backend/core/models.py

```diff
@@ -6,11 +6,9 @@ class ModelRegistry(object):
     def __init__(self):
         self.models = {}
```

Comment:
The `self.models` dictionary is not being updated correctly. The line where the chat model is removed should be reviewed.